### ISSUE TITLE
[Linux] BespokeSynth only works well when compiling in Debug mode.

### ISSUE THREAD

**Issue by xunil-cloud** [BespokeSynth/BespokeSynth#30]
I try to compile from source on Arch Linux. If I compile in **Release** mode, the computer keyboard is not really responsive. For instance, the popup menu won't show up every time when I press 'k'. Another wired thing is that BespokeSynth has no sound in JACK and only has sound in pulseAudio. I also get some messages from terminal:  

```
44.8387: trying to stop before we started (44845<=44845)
45.5121: trying to stop before we started (45517.6<=45517.6)
47.5322: trying to stop before we started (47535.4<=47535.4)
48.2056: trying to stop before we started (48208<=48208)
```
I also try the latest [v0.0004-pre](https://github.com/awwbees/BespokeSynth/releases/tag/v0.0004-pre) binary version, but it has the same issues.

Note: If I compile in **Debug** mode, I have no issue. Extremely AWESOME!

**Comment by djdeath** [comment]
Kind of feels similar to https://github.com/awwbees/BespokeSynth/issues/29

Another thing I noticed for instance is that trying to record the issue with the gnome-shell screen record makes it go away.
I wonder if there is some kind of race condition in the event processing code that would make this happen.
Since you mentioned that a Debug build has no issue, it might just slow things enough to avoid the race... :man_shrugging:

**Comment by xunil-cloud** [comment]
Did you try the keyboarddisplay module? I have similar mouse click issue that you have just described in #29. By the way, I also try a Release build without optimization, but that doesn't fix the problem.

**Comment by awwbees** [comment]
that is very strange! maybe I should release a debug build for linux until I figure it out

**Comment by xunil-cloud** [comment]
I am also trying to figure out why this happened. If I find something new, I will let you know. 🙂

**Comment by awwbees** [comment]
@xunil-cloud if in https://github.com/awwbees/BespokeSynth/blob/master/Source/MainComponent.cpp, you change:
```   static int sRenderFrame = 0;
#if DEBUG
      if (sRenderFrame % 2 == 0)
#else
      if (true)
#endif
      {
         openGLContext.triggerRepaint();
      }
```

to simply

```   static int sRenderFrame = 0;
      if (sRenderFrame % 2 == 0)
      {
         openGLContext.triggerRepaint();
      }
```

does that make it perform like debug? the side effect is that this will limit the display frame rate to 30fps

**Comment by djdeath** [comment]
Having the `if (true)` makes the Debug build behave like the Release one for me (mouse issues).

The debug build works fine here.

**Comment by awwbees** [comment]
@djdeath how about the other way? does the change I recommended make the Release build perform well?

**Comment by djdeath** [comment]
Yep, your change fixes the Release build too.

**Comment by awwbees** [comment]
ah nice! I will update the build with that change, then longer-term, I will try to figure out why linux builds can't handle 60fps. nice to have a solid explanation of why things worked well in Debug but not in Release!

**Comment by awwbees** [comment]
(fix for this was in a0a2a6d9a9307c8b90d37e0dbddd794957a8e528)

**Comment by KottV** [comment]
Must say that I never have such problem on Linux.
Do you use X or Wayland? And what drivers and WM/DE?

**Comment by xunil-cloud** [comment]
I use Xfce and X Window. I have mesa installed on my system. For hardware, I have intel i3 CPU with integrated graphics. Maybe my hardware can't handle 60fps?

**Comment by djdeath** [comment]
I ran under gnome-shell using wayland, also Mesa intel integrated graphics.

**Comment by xunil-cloud** [comment]
Since you say you have no problem with 60fps, I wonder if we need to provide an option to set fps manually, maybe in userprefs.json file.

**Comment by KottV** [comment]
I'm on nvidia with blob and X11/KDE.

**Comment by KottV** [comment]
Ok, I catch this thing. When it happens the output says:

**Xlib:  extension "NV-GLX" missing on display ":0".**

I use build without that fix https://github.com/awwbees/BespokeSynth/commit/a0a2a6d9a9307c8b90d37e0dbddd794957a8e528#diff-8e71264d839b8e1576b1a9c78a82ddd6

But it happen very rarely
